android-components/components/browser/state/src/test/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddlewareTest.kt

```diff
@@ -63,43 +61,52 @@ class TranslationsMiddlewareTest {
     }
 
     @Test
-    fun `WHEN TranslateExpectedAction is dispatched AND fetching languages is successful THEN TranslateSetLanguagesAction is dispatched`() {
-        val supportedLanguages = TranslationSupport(null, null)
+    fun `WHEN OperationRequestedAction is dispatched to fetch languages AND succeeds THEN SetSupportedLanguagesAction is dispatched`() = runTest {
+        val supportedLanguages = TranslationSupport(
+            fromLanguages = listOf(Language("en", "English")),
+            toLanguages = listOf(Language("en", "English")),
+        )
         val callbackCaptor = argumentCaptor<((TranslationSupport) -> Unit)>()
-        val action = TranslationsAction.TranslateExpectedAction(tabId = tab.id)
-        val middlewareContext = mock<MiddlewareContext<BrowserState, BrowserAction>>()
 
-        // Important for ensuring that the tests verify on the same store
```

Comment:
Test seems to be verifying without this. I do wonder if the test is blocking correctly. I see it complete as expected in a debugger, so I think it is. LMK, if something seems off.

Reply:
Refactored this to `setup`. The test seems more stable with that added. I plan on an update on the other test in bug [1878365]( https://bugzilla.mozilla.org/show_bug.cgi?id=1878365).